src/map/ai/controllers/pet_controller.cpp

```diff
@@ -36,21 +36,17 @@ CPetController::CPetController(CPetEntity* _PPet) :
 
 void CPetController::Tick(time_point tick)
 {
-    if (PPet->isCharmed && tick > PPet->charmTime)
+    if ((PPet->isCharmed && tick > PPet->charmTime) || ((PPet->PMaster == nullptr) && PPet->isAlive()))
     {
         petutils::DespawnPet(PPet->PMaster);
```

Comment:
You deleted DespawnPet in petutils, does this even compile??

Reply:
WIP, I'll have to fix this later